src/main/java/org/codehaus/mojo/versions/DependencyUpdatesReport.java

```diff
@@ -129,10 +130,24 @@ protected void doGenerateReport( Locale locale, Sink sink )
                 // Using the original model to get the original dependencyManagement entries and
                 // not the interpolated model.
                 // TODO: I'm not 100% sure if this will work correctly in all cases.
+
+                // Grab the project properties
+                Properties props = getProject().getProperties();
                 for ( Dependency dep : getProject().getOriginalModel().getDependencyManagement().getDependencies() )
                 {
+                    String version = dep.getVersion();
                     getLog().debug( "Original Dpmg: " + dep.getGroupId() + ":" + dep.getArtifactId() + ":"
-                        + dep.getVersion() + ":" + dep.getType() + ":" + dep.getScope() );
+                        + version + ":" + dep.getType() + ":" + dep.getScope() );
+                    // if the version is referencing a property, dereference it
+                    if ( version.startsWith( "${" ) )
```

Comment:
what about versions which are made from more complex strings? e.g. `1.${oldVersion}-SNAPSHOT`?

Reply:
Interesting thought. I've not seen that used in the wild, and the current plugin codebase doesn't suggest that it's a supported approach:

https://github.com/mojohaus/versions-maven-plugin/blob/b17a9271478abc12fd33b36d443f476f60005109/src/main/java/org/codehaus/mojo/versions/AbstractVersionsDependencyUpdaterMojo.java#L192